pkg/controller/controller.go

```diff
@@ -462,42 +479,50 @@ func convertCatalog(in *osb.CatalogResponse) ([]*v1alpha1.ServiceClass, error) {
 			if err != nil {
 				err = fmt.Errorf("Failed to marshal metadata\n%+v\n %v", svc.Metadata, err)
 				glog.Error(err)
-				return nil, err
+				return nil, nil, err
 			}
-			ret[i].ExternalMetadata = &runtime.RawExtension{Raw: metadata}
+			serviceClasses[i].ExternalMetadata = &runtime.RawExtension{Raw: metadata}
 		}
 
-		ret[i].SetName(svc.Name)
+		serviceClasses[i].SetName(svc.Name)
+
+		// set up the plans using the ServiceClass Name
+		plans, err := convertServicePlans(svc.Plans, svc.Name)
+		if err != nil {
+			return nil, nil, err
+		}
+		servicePlans = append(servicePlans, plans...)
 	}
-	return ret, nil
+	return serviceClasses, servicePlans, nil
 }
 
-func convertServicePlans(plans []osb.Plan) ([]v1alpha1.ServicePlan, error) {
-	ret := make([]v1alpha1.ServicePlan, len(plans))
-	for i := range plans {
-		ret[i] = v1alpha1.ServicePlan{
-			Name:        plans[i].Name,
-			ExternalID:  plans[i].ID,
-			Free:        (plans[i].Free != nil && *plans[i].Free),
-			Description: plans[i].Description,
+func convertServicePlans(plans []osb.Plan, serviceClassName string) ([]*v1alpha1.ServicePlan, error) {
```

Comment:
I believe that we will need to prefix the name of the service plan with the name of the service class.  Plan names are only unique within a service, and not globally.  I don't think this yields the greatest names, but we can fix that with porcelain.

Totally open to talking through this in a meeting, and I realize it is an intersection with an unresolved issue (#802).

Reply:
Fix it later.

Reply:
we should discuss this before beta.  Morgan can you open an issue and put it in the 0.1.0 milestone so we don't forget.

Reply:
okay

Reply:
Circling back in this group of comments for posterity:

We decided in a SIG meeting for now to use `$name-$id` as the kubernetes names of services to spur ourselves to finish figuring out what to do with the kubernetes names of these objects.